### PR TITLE
JAMES-3962 JMAP Email/set: specific headers for body parts

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala [hunk 1]
```diff
@@ -73,9 +73,21 @@ case class ClientPartId(id: Id)
 
 case class ClientBody(partId: ClientPartId, `type`: Type)
 
+case class ClientEmailBodyValueWithoutHeaders(value: String,
+                                isEncodingProblem: Option[IsEncodingProblem],
+                                isTruncated: Option[IsTruncated]) {
+  def withHeaders(specificHeaders: List[EmailHeader]): ClientEmailBodyValue = {
+    println(specificHeaders)
```
Comment (quantranhong1999) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala:80]: debug

**Review by quantranhong1999 — APPROVED** [review]

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala [hunk 2]
```diff
@@ -3171,6 +3171,110 @@ trait EmailSetMethodContract {
...
+           |                                    "type": "text/plain"
+           |                                },
+           |                                {
+           |                                    "header:Specific:asText": "MATCHME",
+           |                                    "type": "text/html"
+           |                                }
+           |                            ],
+           |                            "header:Specific:asText": null,
+           |                            "type": "multipart/alternative"
+           |                        }
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala:3274]: indent

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailSetSerializer.scala [hunk 3]
```diff
@@ -407,27 +436,10 @@ class EmailSetSerializer @Inject()(messageIdFactory: MessageId.Factory, mailboxI
...
-              .flatMap(specificHeaderRequest => asReads(specificHeaderRequest)
-                .reads(value).asEither.left.map(e => new IllegalArgumentException(e.toString()))
-                .map(headerValue => EmailHeader(EmailHeaderName(specificHeaderRequest.headerName), headerValue)))
-        }.sequence
-
-        specificHeadersEither.fold(e => JsError(e.getMessage),
-          specificHeaders => withoutHeader.map(_.toCreationRequest(specificHeaders)))
+        extractSpecificHeaders(o)
+          .fold(e => JsError(e.getMessage),
+          specificHeaders => emailCreationRequestWithoutHeadersReads.reads(o).map(_.toCreationRequest(specificHeaders)))
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailSetSerializer.scala:441]: indent?

**Review by Arsnael — APPROVED** [review]
Just small remarks but overall I approve :)

**Comment by Arsnael** [comment]
```
09:34:32,937 [ERROR] Failed to execute goal io.github.evis:scalafix-maven-plugin_2.13:0.1.6_0.9.34:scalafix (scala-check-style) on project james-server-jmap-rfc-8621: Scalafix invoked with errors. Check logs for details. -> [Help 1]
```

**Review by Arsnael — APPROVED** [review]

**Pull request merged**